litellm/caching.py

```diff
@@ -1446,12 +1449,389 @@ def delete_cache(self, key):
+
+    def _get_cache_logic(self, cached_response: Any):
+        """
+        Common 'get_cache_logic' across sync + async redis client implementations
+        """
+        if cached_response is None:
+            return cached_response
+
+        # check if cached_response is bytes
+        if isinstance(cached_response, bytes):
+            cached_response = cached_response.decode("utf-8")
+
+        try:
+            cached_response = json.loads(
+                cached_response
+            )  # Convert string to dictionary
+        except:
+            cached_response = ast.literal_eval(cached_response)
+        return cached_response
+
+    def set_cache(self, key, value, **kwargs):
+        import numpy as np
+
+        print_verbose(f"dual-semantic-cache (sync) set_cache, kwargs: {kwargs}")
+
+        messages = kwargs["messages"]
+        prompt = "".join(message["content"] for message in messages)
+
+        embedding_response = litellm.embedding(
+            model=self.embedding_model_name,
+            input=prompt,
+            cache={"no-store": True, "no-cache": True},
+        )
+
+        # get the embedding
+        embedding = embedding_response["data"][0]["embedding"]
+
+        # make the embedding a numpy array, convert to bytes
+        embedding_bytes = np.array(embedding, dtype=np.float32).tobytes()
+
+        if self.in_memory_cache is not None:
+            if "ttl" not in kwargs and self.in_memory_default_ttl is not None:
+                kwargs["ttl"] = self.in_memory_default_ttl
+
+            self.in_memory_cache.set_cache(key, value, **kwargs)
+        else:
+            print_verbose(
+                "dual-semantic-cache (sync) set_cache: in_memory_cache is None"
+            )
+
+        value = str(value)
+        assert isinstance(value, str)
+
+        self.index.load(
+            [
+                {
+                    "response": value,
+                    "prompt": prompt,
+                    "litellm_embedding": embedding_bytes,
+                }
+            ]
+        )
+
+    def get_cache(self, key, **kwargs):
+        print_verbose(f"dual-semantic-cache (sync) get_cache, kwargs: {kwargs}")
+
+        if self.in_memory_cache is not None:
+            in_memory_result = self.in_memory_cache.get_cache(key, **kwargs)
+
+            if in_memory_result is not None:
+                print_verbose("dual-semantic-cache (sync) in-memory-cache hit")
+                return in_memory_result
+            else:
+                print_verbose(
+                    "dual-semantic-cache (sync) get_cache: in-memory-cache miss"
+                )
+
+        else:
+            print_verbose(
+                "dual-semantic-cache (sync) get_cache: in_memory_cache is None"
+            )
+
+        from redisvl.query import VectorQuery
+
+        messages = kwargs["messages"]
+        prompt = "".join(message["content"] for message in messages)
+
+        embedding_response = litellm.embedding(
```

Comment:
is this making an embedding call inside caching which itself could be called inside an embedding call?

Reply:
no, because these calls set "no-cache" to True